include/sparrow/layout/array_base.hpp

```diff
@@ -101,6 +115,22 @@ namespace sparrow
         const_bitmap_range bitmap() const;
         const_value_range values() const;
 
+        void resize(size_type new_size, const value_type& value);
+
+        iterator insert(const_iterator pos, const value_type& value);
+        iterator insert(const_iterator pos, const value_type& value, size_type count);
+        iterator insert(const_iterator pos, std::initializer_list<value_type> values);
+        template <typename InputIt>
+        iterator insert(const_iterator pos, InputIt first, InputIt last);
+        template <std::ranges::input_range R>
+        iterator insert(const_iterator pos, const R& range);
+
+        iterator erase(const_iterator pos);
+        iterator erase(const_iterator first, const_iterator last);
+
+        void push_back(const value_type& value);
+        void pop_back();
```

Comment:
I think having these methods in public might be "dangerous" as not all the arrays inheriting from the CRTP implement these methods. Having them in private + using declarations in the inheriting classes implementing this API would be safer.

On the long run it would be better to have them in a dedicated CRTP which arrays with modifiers can inherit from, but this can be tricky to implement because of the bitmap CRTP.

Reply:
New crtp classes (with bitmap and mutable) are created